app/assets/javascripts/models/ui/state_manager.coffee

```diff
@@ -251,6 +251,8 @@
       if @_isValid(path, serialized)
         changed = urlUtil.saveState(path, serialized, !@historyChanged, @page.workspaceNameField.peek())
         @historyChanged = true if changed
+        if path == '/search/collection-details' || path == '/search/granules' || path == '/search/project'
```

Comment:
I think this is the wrong logic. page.collections should not be relevant on any of these paths. It should only be relevant when the collections view is visible. It's also not necessarily relevant when transitioning *from* these paths. For example, from the project page, you can transition to collection details or granules, which should not make the collection search relevant.